drv/cosmo-hf/src/main.rs

```diff
@@ -259,6 +257,9 @@ impl FlashDriver {
     }
 
     /// Writes data from a `BufReader` into the flash
+    ///
+    /// This function will only return an error if it fails to write to a
```

Comment:
**praise:** Good addition.

I wonder if it would be a good worth it to make an explicit infallible version to avoid the need for `.unwrap_lite()`? Would that help remove some unnecessary codegen?